src/headers/tomcrypt_pk.h

```diff
@@ -254,9 +254,6 @@ int dh_check_pubkey(dh_key *key);
 
 /** Structure defines a GF(p) curve */
 typedef struct {
-   /** The size of the curve in octets */
-   int size;
```

Comment:
@karel-m What is the reason for removing this?

Reply:
To me it looks like it's not needed as a user will normally not identify the curve by its size but by its name.
Also the size got ambiguous now that there are multiple standard-curves with the same size.

But that's just a wild guess :)
@karel-m ?

Reply:
Hmm, I think that keeping the size or rather, changing the size to refer to the curve size in bits would be better. Many crypto libraries allow one to just specify a requested curve size in bits and requested field type and then provide a default named curve if one is present. (For example Java Cryptographic Architecture requires that)

Removing the size or storing the byte size can also pose problems for curves of bit size `% 8 != 0`.

Reply:
There's still `ecc_make_key()` which takes the size of the desired key and then chooses one for you :)

Reply:
`ecc_make_key()` currently works by looking at the `size` attribute of curves.

This PR removes that attribute and does the lookup at [ecc_set_dp_size](https://github.com/libtom/libtomcrypt/pull/236/files#diff-72b3a87b26a1dfde185685298ed58b33R51) by hardcoding the values.

Reply:
> This PR removes that attribute and does the lookup at ecc_set_dp_size by hardcoding the values.

yeah, it's truly not elegant at all... :)

I think it'd make sense to add a `ecc_get_dp_by_bits` (or sth like that) which finally uses bits instead of octets, then iterate there over the `ltc_ecc_sets` array, and compare the `mp_count_bits(key->dp.prime)`. `ecc_make_key()` would then call that one with `keysize*8`

Reply:
The `int size` is simply redundant (it is the byte size of `prime`). It used to be a unique identifier of elliptic curve but it is not true any more. For some sizes we several curves so it does not make sense to ask for a curve just by giving a size (no matter whether byte size or bit size). You should use curve name or curve OID. 

The `ecc_set_dp_size` was implemented just for compatibility reasons. It is not nice but it is a backward compatible mapping between old and new API.